comptests/s3/test_rest_api_rgw.py

```diff
@@ -256,16 +333,645 @@ def test_user_create_36665(self):
+            'max-buckets': max_buckets
+        }
+        self.log.info(
+            "Step 1: Creating a new IAM user with name %s", str(user_name))
+        loop = asyncio.get_event_loop()
+        status, user_info = loop.run_until_complete(self.obj.create_user(user_params))
+        assert status == HTTPStatus.OK, "Not able to create user. Test Failed"
+        self.log.info("Created user details: %s", user_info)
+        self.log.info(
+            "Step 2: Verifying that new IAM user is created successfully")
+        status, user_info = loop.run_until_complete(self.obj.get_user_info(user_params))
+        assert status == HTTPStatus.OK, "Not able to Get user Info. Test Failed"
+        self.log.info("Get user info output: %s", user_info)
+        self.created_users.append(user_params)
+        self.log.info(
+            "Step 3: Try creating a bucket as max_buckets is set to negative value.")
+        self.user_info = json.loads(user_info)
+        self.access_key_io = self.user_info['keys'][0]['access_key']
+        self.secret_key_io = self.user_info['keys'][0]['secret_key']
+        self.log.info("Credentials : Access key = %s , secret key = %s",
+                      self.access_key_io, self.secret_key_io)
+        self.io_obj = S3TestLib(access_key=self.access_key_io,
+                                secret_key=self.secret_key_io)
+        self.bucket_list = []
+        try:
+            self.bucket_name = f"{self.bucket_name_prefix}" \
+                               f"{str(time.perf_counter_ns()).replace('.', '_')}"
+            self.log.info("Started creating bucket")
+            resp = self.io_obj.create_bucket(bucket_name=self.bucket_name)
+            if resp[0]:
+                self.log.info("Failed : Bucket got created :%s , but it should not get created "
+                              "as max_bucket is set to negative value : %s", resp[1], max_buckets)
+                self.bucket_list.append(resp[1])
+                assert False, "Able to create buckets even with negative value for max_buckets"
+        # pylint: disable=broad-except
+        except BaseException as err:
+            self.log.warning("Step 4: As expected bucket creation got failed when "
+                             "negative value is set for max_buckets, with error: %s", err)
+        self.log.info("END : %s", test_case_name)
+
+    @pytest.mark.api_user_ops
+    @pytest.mark.tags('TEST-37155')
+    def test_user_create_37155(self):
+        """Test create iam user specifying tenant."""
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("START: Test create iam user specifying tenant.")
+        user_name = f"{self.user_name_prefix}{str(time.perf_counter_ns()).replace('.', '_')}"
+        tenant = "tnt1"
+        user_params = {
+            'display-name': user_name,
+            'uid': user_name,
+            'tenant': tenant
```

Comment:
I am not sure of the test behavior here. Is it expected that tenant of this name was created ?